core/scheduler/scheduler_test.go

```diff
@@ -202,21 +201,11 @@ func TestSchedulerDuties(t *testing.T) {
 			eth2Cl, err := beaconmock.New(
 				beaconmock.WithValidatorSet(valSet),
 				beaconmock.WithGenesisTime(t0),
-				beaconmock.WithDeterministicDuties(test.Factor),
+				beaconmock.WithDeterministicAttesterDuties(test.Factor),
+				beaconmock.WithDeterministicProposerDuties(test.Factor),
 			)
 			require.NoError(t, err)
 
-			// Wrap ProposerDuties to returns some errors
-			origFunc := eth2Cl.ProposerDutiesFunc
-			eth2Cl.ProposerDutiesFunc = func(ctx context.Context, epoch eth2p0.Epoch, indices []eth2p0.ValidatorIndex) ([]*eth2v1.ProposerDuty, error) {
-				if test.PropErrs > 0 {
-					test.PropErrs--
-					return nil, errors.New("test error")
-				}
-
-				return origFunc(ctx, epoch, indices)
-			}
```

Comment:
this actually tested retry of proposer duty fetching, now you are not testing this anymore...